### DISCUSSION TITLE
ssh-agent: program always prompts for passphrase

### DISCUSSION THREAD

**Discussion by rieje** [netblue30/firejail#5845] [Q&A] [answered]
I successfully set up ssh-agent but it only works when I don't use the firejail version (i.e. /usr/bin/ssh). If I use firejail ssh, it will just prompt for passphrase as if I'm ssh-agent is not used.

When I set up ssh-agent, the it also involved defining the env variables `SSH_AUTH_SOCK="$XDG_RUNTIME_DIR/ssh-agent.socket"` and `SSH_ASKPASS="lxqt-openssh-askpass"` and uses `/usr/bin/ssh-agent` and `/usr/bin/lxqt-openssh-askpass`. I tried whitelisting/private-bin the paths/variables but it results in `Error: no suitable /usr/bin/ssh executable found`.

EDIT: Solved with `whitelist ${RUNUSER}/ssh-agent.socket`

**Comment by ghost** [comment]
Some related threads that might provide more insights/things you can try:

- [ssh with gpg-agent](https://github.com/netblue30/firejail/discussions/4869)
- discussion in [ssh-agent: hardening](https://github.com/netblue30/firejail/pull/5753)

Please provide more detailed examples and logs of your attempts with firejailing ssh-agent. We're pretty blind without those. Or consider creating an issue from this discussion.

**Comment by kmk3** [comment, marked as answer]
@rieje on [Jun 3, 2023](https://github.com/netblue30/firejail/discussions/5845#discussion-5261523):

> EDIT: Solved with `whitelist ${RUNUSER}/ssh-agent.socket`
